ui/perfherder/CompareTable.jsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import { Table, Progress } from 'reactstrap';
+import PropTypes from 'prop-types';
+import Icon from 'react-fontawesome';
+
+import SimpleTooltip from '../shared/SimpleTooltip';
+
+import { displayNumber } from './helpers';
+import TableAverage from './TableAverage';
+
+export default class CompareTable extends React.Component {
+  getColorClass = (data, type) => {
+    const { className, isRegression, isImprovement } = data;
+    if (type === 'bar' && !isRegression && !isImprovement) return 'secondary';
+    if (type === 'background' && className === 'warning')
+      return `bg-${className} ${
+        className !== 'warning' ? 'text-white font-weight-bold' : null
+      }`;
+    if (type === 'text' && className) return `text-${className}`;
+    return className;
+  };
+
+  deltaTooltipText = (delta, percentage, improvement) =>
+    `Mean difference: ${displayNumber(delta)} (= ${Math.abs(
+      displayNumber(percentage),
+    )}% ${improvement ? 'better' : 'worse'})`;
+
+  render() {
+    const { data, testName, title } = this.props;
+    return (
+      <Table sz="small" className="compare-table mb-0" key={title || testName}>
+        <thead>
+          <tr
+            style={{ backgroundColor: 'lightgrey' }}
```

Comment:
Inline styles don't work with CSP (bug 1270157) - could you convert these to CSS classes whilst you're here? (Some will be supported by bootstrap utilities so may not need new class definitions)

Reply:
I've been using inline styles in all new Perf components, so I'll open another bug to change those to classes.